hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java

```diff
@@ -2472,4 +2482,13 @@ boolean snapshotCleanupSwitch(final boolean on, final boolean synchronous)
    */
   void updateRSGroupConfig(String groupName, Map<String, String> configuration) throws IOException;
 
+  /**
+   * Retrieve recent online records from HMaster / RegionServers.
+   * Examples include slow/large RPC logs, balancer decisions by master.
+   *
+   * @param logRequest request payload with possible filters
+   * @return Log entries representing online records from servers
+   * @throws IOException if a remote or network exception occurs
+   */
+  List<LogEntry> getLogEntries(LogRequest logRequest) throws IOException;
```

Comment:
This needs an optional parameter to allow constrained clients to limit the size of the returned list of LogEntry.

Reply:
Please define a limit parameter for the Admin API. I'm open to other suggestions, but by "individual use cases"  I believe you mean the RPC message classes, and those are not public API.

As an alternative you can do the same thing as you did for the SlowLog API where a filter can be supplied, and one thing the filter interface lets you do is specify a limit.

I won't approve this without a limit option of some kind in the public user facing admin API.

Reply:
This is our Admin endpoint:
```
List<LogEntry> getLogEntries(LogRequest logRequest) throws IOException;
```
Now when client uses this to get slowLog response, they can provide `limit` in specific payload which extends `LogRequest` and send it over, similarly they can provide `limit` in BalancerDecisionRequest which extends `LogRequest` and get the response accordingly, both request payloads that extend `LogRequest` are anyways public interfaces for clients. Now, client would rather provide `limit` in request payload than as separate argument on this Admin endpoint (method signature). Does that sound good?